Sources/Core/AWSClientRuntime/Endpoints/AWSEndpoint.swift

```diff
@@ -70,4 +75,48 @@ extension Endpoint {
         
         return scheme
     }
-}
+
+    /// Returns the first auth scheme
+    /// This is an internal API and subject to change without notice
+    /// - Returns: first auth scheme if present
+    public func firstAuthScheme() -> [String: Any]? {
```

Comment:
Nit: Removing this function and replacing it with a function that returns authSchemes would help reduce key duplication (the `authScheme(name: String)` function also accesses authSchemes), and since the return type is an array, callers already know how to get the first item in an array so no need for an explicit function to do so. 

```swift
public func authSchemes() -> [[String: Any]]? {
    return properties["authSchemes"] as? [[String: Any]]
}

let firstAuthScheme = authSchemes()?.first
```

Reply:
> Looks great! Love the docs and tests!
> 
> imo it'd be nice to have the dictionary keys defined in some embedded type. but not required and something we can always do in the future.
> 
> ```swift
> enum PropertyKeys {
>     static let authSchemes = "authSchemes"
> }
> 
> enum AuthSchemeKeys {
>     static let signingRegion = "signingRegion"
>     static let signingRegionSet = "signingRegionSet" 
>     // etc...
> }
> ```

adapted to suggestion